### PR TITLE
Takedown bow and multi-function quiver

### PR DESCRIPTION
**Pull request by Termineitor244**
#### Summary
Content "Adds a new bow and a quiver/bag that can hold it too"

#### Purpose of change
I found the items online and I wanted to see them in game, they are cool and would be useful to have for an archer character.

#### Describe the solution
Adds a new recursive bow, the takedown bow (Which for now works the same as the normal recursive bow, I don't know enough about these items to make it different), as well as a new quiver that can hold some arrows, a takedown bow and even some other items!

#### Describe alternatives you've considered
To make the takedown bow need to transform for easy storage, right now its max length is for its "taken down" form.

Edit: Done! You need to make it transform for easier storage!

#### Testing
It works! The item spawns, fires, transforms, and the quiver can hold arrows, the bow, and even other items!

#### Additional context

<details> 
<summary>Image and link of the item in real life:</summary>

https://www.amazon.com/XTACER-Multi-Function-4-Tubes-Training-Takedown/dp/B072K4537K?th=1
imagen

</details>

### CODE REVIEW

**Comment by Sathra225** [comment]
Ah, so you'd want a use_action transform like the extendable baton. Then a pocket restriction for 'folded_takedown'. Could have the folded version with BELT_CLIP as well (if you can do it with a bat, a folded bow as in the picture could as well).

**Comment by Termineitor244** [comment]
> Ah, so you'd want a use_action transform like the extendable baton. Then a pocket restriction for 'folded_takedown'. Could have the folded version with BELT_CLIP as well (if you can do it with a bat, a folded bow as in the picture could as well).

I had originally made it like that, but wouldn't there be a problem with the transform since you can load your bow with one arrow? That's why I decided to remove the folded version and just reduce the length of the main item.

**Comment by catdach** [comment]
> > Ah, so you'd want a use_action transform like the extendable baton. Then a pocket restriction for 'folded_takedown'. Could have the folded version with BELT_CLIP as well (if you can do it with a bat, a folded bow as in the picture could as well).
> 
> I had originally made it like that, but wouldn't there be a problem with the transform since you can load your bow with one arrow? That's why I decided to remove the folded version and just reduce the length of the main item.

Actually, no. One of the weird things about bows in game is that you can't load them without firing them. I think it's because of the "RELOAD_TO_SHOT" flag? If you spawn a bow in via debug menu it will be in a "loaded" state; but after you fire it, you cant actually return it to that state. 

However, one thing to note is that takedown bows actually become quite a few pieces when "taken down" [sorce](https://www.amazon.com/TOPARCHERY-Archery-Takedown-Hunting-Recurve/dp/B06WPB9DMS/ref=sr_1_15_sspa?c=ts&keywords=Archery%2BBows&qid=1651349782&s=hunting-fishing&sr=1-15-spons&ts_id=3395521&smid=A333LXZRU4FUXK&spLa=ZW5jcnlwdGVkUXVhbGlmaWVyPUEyRUU4NENHWlo0TU9CJmVuY3J5cHRlZElkPUExMDIyNjI4MjVTRElSRUdBTUVKUyZlbmNyeXB0ZWRBZElkPUEwOTA4MzI4S1U4MFdXU0tGREY2JndpZGdldE5hbWU9c3BfbXRmJmFjdGlvbj1jbGlja1JlZGlyZWN0JmRvTm90TG9nQ2xpY2s9dHJ1ZQ&th=1)
image
I don't know if you can justify clipping this many pieces to a belt (presumably that's why this special quiver for it would exist). It's also a _fairly_ lengthy process to dis- and re- assemble (a couple minutes maybe?) So if it becomes a "foldable" item, the transformation time should probably take a while.

**Comment by Termineitor244** [comment]
I will make a foldable transformation then, taking some time of course. In the meanwhile I will leave this in draft.

About times... I will se videos on youtube of people using these kinds of bows and try to guide the time from that (A video I just saw  had a guy doing it in a minute).

**Comment by Termineitor244** [comment]
I made the folded version and modified what needed to be modified but.. Well... The bow needs to be loaded to transform (Not enough charges)... But it's a bow...

I tried with `"need_charges": 0` but it does not seems to be working... I will ask if someone with C++ knowledge wants to help with this.

File: data/json/itemgroups/Clothing_Gear/clothing.json [hunk 1]
```diff
@@ -2900,6 +2903,7 @@
       [ "dump_pouch", 6 ],
       [ "ragpouch", 22 ],
       [ "leather_pouch", 16 ],
+      [ "quiver_takedown_bow", 5 ],
```
Comment (kevingranade) [inline data/json/itemgroups/Clothing_Gear/clothing.json:2906]: Redundant entry, and it makes it too common.
```suggestion
```

File: data/json/itemgroups/Clothing_Gear/clothing.json [hunk 2]
```diff
@@ -3125,6 +3129,7 @@
       [ "antarvasa", 1 ],
       [ "uttarasanga", 1 ],
       [ "samghati", 1 ],
+      [ "quiver_takedown_bow", 1 ],
```
Comment (kevingranade) [inline data/json/itemgroups/Clothing_Gear/clothing.json:3132]: This is too common an item group to put something this obscure in.
```suggestion
```

File: data/json/itemgroups/Clothing_Gear/clothing.json [hunk 3]
```diff
@@ -3573,6 +3578,7 @@
           { "item": "case_violin", "prob": 5 },
           { "item": "bookstrap", "prob": 5 },
           { "item": "camera_bag", "prob": 20 },
+          { "item": "quiver_takedown_bow", "prob": 20 },
```
Comment (kevingranade) [inline data/json/itemgroups/Clothing_Gear/clothing.json:3581]: Way too common.
```suggestion
          { "item": "quiver_takedown_bow", "prob": 2 },
```

File: data/json/itemgroups/Locations_MapExtras/mansion.json [hunk 4]
```diff
@@ -849,6 +849,8 @@
       [ "compbow", 1 ],
       [ "shortbow", 5 ],
       [ "recurbow", 2 ],
+      [ "takedown_recurbow", 2 ],
+      [ "quiver_takedown_bow", 2 ],
```
Comment (kevingranade) [inline data/json/itemgroups/Locations_MapExtras/mansion.json:853]:
```suggestion
      [ "takedown_recurbow", 1 ],
      [ "quiver_takedown_bow", 1 ],
```

File: data/json/itemgroups/SUS/lodge.json [hunk 5]
```diff
@@ -9,6 +9,8 @@
       { "item": "compbow_high", "prob": 20 },
       { "item": "compbow_low", "prob": 40 },
       { "item": "recurbow", "prob": 30 },
+      { "item": "takedown_recurbow", "prob": 30 },
+      { "item": "quiver_takedown_bow", "prob": 10 },
```
Comment (kevingranade) [inline data/json/itemgroups/SUS/lodge.json:13]:
```suggestion
      { "item": "takedown_recurbow", "prob": 3 },
      { "item": "quiver_takedown_bow", "prob": 1 },
```

File: data/json/itemgroups/activities_hobbies.json [hunk 6]
```diff
@@ -274,6 +275,8 @@
       [ "arrowrest", 10 ],
       [ "shortbow", 5 ],
       [ "recurbow", 2 ],
+      [ "takedown_recurbow", 2 ],
+      [ "quiver_takedown_bow", 2 ],
```
Comment (kevingranade) [inline data/json/itemgroups/activities_hobbies.json:279]:
```suggestion
      [ "takedown_recurbow", 1 ],
      [ "quiver_takedown_bow", 1 ],
```

File: data/json/itemgroups/main.json [hunk 7]
```diff
@@ -242,6 +242,8 @@
       { "item": "compbow", "prob": 40 },
       { "item": "shortbow", "prob": 20 },
       { "item": "recurbow", "prob": 40 },
+      { "item": "takedown_recurbow", "prob": 40 },
+      { "item": "quiver_takedown_bow", "prob": 10 },
```
Comment (kevingranade) [inline data/json/itemgroups/main.json:246]:
```suggestion
      { "item": "takedown_recurbow", "prob": 2 },
      { "item": "quiver_takedown_bow", "prob": 1 },
```

**Comment by Termineitor244** [comment]
Thanks for the review! I'm not quite sure about the low probability of the bow in bow related item groups (For what I see, it seems a good portion of sports related bows are takedown bows), but I accept the change nonetheless.

**Pull request merged**